lib/signex/parameters.ex

```diff
@@ -0,0 +1,48 @@
+defmodule SignEx.Parameters do
+  @moduledoc """
+    A `SignEx.Parameters` struct contains all the information needed to reconstruct the signature of a message.
+  """
+
+  require Logger
+
+  defstruct [:key_id, :algorithm, :headers, :signature]
+
+  def to_string(params) do
+    Logger.warn("Deprechiated: Use `SignEx.Parameters.serialize`")
+    serialize(params)
+  end
+
+  @doc """
+  Serialize parameters as a HTTP signature header value
+  """
+  def serialize(%__MODULE__{
+    key_id: key_id,
+    algorithm: algorithm,
+    headers: headers,
+    signature: signature
+    }) do
+      headers = Enum.join(headers, " ")
+     {:ok, "key_id=\"#{key_id}\",algorithm=\"#{algorithm}\",headers=\"#{headers}\",signature=\"#{signature}\""}
```

Comment:
I don't think this function should return the auth header. I think it should return the serialized parametes. because this is exactly the string you'd want it we ever did start putting stuff in the signature headers. Also if we have more than one authorisaztion method I like using the following
```elixir
case auth_header do
  "Signature " <> serializer_params ->
    parse(serialized_params) |> # and more stuff
  "Other " <> details ->
    #etc
end